docs/data/data-grid/column-menu/column-menu.md

```diff
@@ -0,0 +1,138 @@
+
+```tsx
+function CustomColumnMenu(props: GridColumnMenuProps) {
+  return (
+    <GridColumnMenuDefault
+      {...props}
+      components={{
+        // Override slot for `ColumnMenuFilterItem`
+        ColumnMenuFilterItem: CustomFilterItem,
+        // Hide `ColumnMenuColumnsItem`
+        ColumnMenuColumnsItem: null,
+        // Add new item
+        ColumnMenuUserItem: CustomUserItem,
+      }}
+      componentsProps={{
+        // Swap positions of filter and sort items
+        columnMenuFilterItem: {
+          displayOrder: 0, // Previously `10`
+        },
+        columnMenuSortItem: {
+          displayOrder: 10, // Previously `0`
+        },
+        columnMenuUserItem: {
+          // set `displayOrder` for new item
+          displayOrder: 15,
+          // pass additional props
+          myCustomValue: 'Do custom action',
+          myCustomHandler: () => alert('Custom handler fired'),
+        },
+      }}
+      initialItems={['ColumnMenuUserItem']}
+    />
+  );
+}
+```
+
+{{"demo": "ReuseColumnMenuGrid.js", "bg": "inline"}}
+
+#### Display orders for default items
+
+As a reference, here is the order for each of the default items, there are some gaps to be able to place items in between:
+
+| **Component**             | **Package** | **Display order (default design)** | **Display order (simple design)** |
+| ------------------------- | ----------- | ---------------------------------: | --------------------------------: |
+| ColumnMenuSortItem        | Community   |                                  0 |                                 0 |
+| ColumnMenuFilterItem      | Community   |                                 10 |                                10 |
+| ColumnMenuHideItem        | Community   |                                 20 |                                20 |
+| ColumnMenuColumnsItem     | Community   |                                 30 |                                30 |
+| ColumnMenuPinningItem     | Pro         |                                  5 |                                35 |
+| ColumnMenuAggregationItem | Premium     |                                 17 |                                37 |
+| ColumnMenuGroupingItem    | Premium     |                                 13 |                                33 |
+
+:::info
+**Tip:** To hide/remove a default item, you can simply pass the component in `components` as `null`
+:::
+
+### Custom component
+
+To customize even more like adding custom items you can opt not to use the default column menu component and pass a totally new column menu component using `ColumnMenu` grid slot.
+
+{{"demo": "CustomColumnMenuGrid.js", "bg": "inline"}}
+
+:::info
+To conditionally render some items, you can use ColDef properties on the `currentColumn` recieved in the props.
```

Comment:
I'm not sure I understand this statement. Do you mean showing column items only for some columns?

Reply:
Yes, basically use `column.field === 'x'` to hide or show specific menu items